pkg/resolver/framework/fakecontroller.go

```diff
@@ -0,0 +1,65 @@
+/*
+ Copyright 2022 The Tekton Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package framework
```

Comment:
I had to put the fake controller/resolver stuff in this package in order to use them in `reconciler_test.go` thanks to the eternal fun of circular dependencies, but the fake controller in particular is intended to be used by real `Resolver` implementations, and the fake resolver will (eventually!) end up being used in Pipeline's unit tests once we merge Resolution into Pipeline.

Reply:
Also, @chitrangpatel, I think this and `fakeresolver.go` may be helpful in writing tests for `pkg/resource/crd_resource.go`.